compiler/rustc_index/src/vec.rs

```diff
@@ -65,7 +65,7 @@ impl Idx for u32 {
 /// `u32::MAX`. You can also customize things like the `Debug` impl,
 /// what traits are derived, and so forth via the macro.
 #[macro_export]
-#[allow_internal_unstable(step_trait, rustc_attrs)]
+#[allow_internal_unstable(step_trait, rustc_attrs, trusted_step)]
```

Comment:
Can you explain what this does? I don't know how allow_internal_unstable works.

Reply:
allow_internal_unstable allows a macro to use a feature gate without having to enable the feature gate in the target crate. This is for example used by libstd to allow `panic!()` to call an unstable function to start panicking without the user having to enable the respective feature gate.

Reply:
Ok, that seems fine in the compiler itself.

Reply:
Wait, how does that work? Aren't feature gates checked after macro expansion usually?

Reply:
The expansion data retrievable based on the expansion id stored in a span contains a field with all `allow_internal_unstable` feature gates set for the expanding macro. `span.allows_unstable(feature)` checks this list.